torch/jit/_recursive.py

```diff
@@ -117,19 +117,48 @@ def recursive_script(mod):
         if not _jit_internal.is_ignored_fn(mod.forward):
             methods = ('forward',)
     exported = []
+    overloads = []
     for name in dir(mod):
         item = getattr(mod, name)
         if callable(item):
             if _jit_internal.get_torchscript_modifier(item) is _jit_internal.FunctionModifiers.EXPORT:
                 exported.append(name)
+
+            # builtin functions like repr() in python 2 do not have __module__ defined
+            if hasattr(item, "__module__") and item.__module__ is not None:
+                method_overloads = _jit_internal._get_overloaded_methods(item, mod._get_name())
+                if method_overloads is not None:
+                    overloads.append((item, method_overloads))
+
     methods = methods + tuple(exported)
 
+    overload_name_mappings = dict(getattr(mod, "__overloads__", {}))
+    overload_stubs = []
+
+    for orig_fn, overload_fns in overloads:
+        orig_ast = torch.jit.get_jit_def(orig_fn, self_name="ScriptModule")
+        names = list(map(lambda i: orig_ast.name().name + "__" + str(i), range(len(overload_fns))))
```

Comment:
Since the overloaded methods aren't directly callable why do they need to be added as normal methods with this mangled name to the `ScriptModule`? The class could hold its schemas for each overload and check that in `OverloadedMethodValue` instead of what it's doing now

Reply:
The normal name isn't being added, only the mangled ones are being added. 

Defining the mangled ones lazily would be an improvement but I don't think it's necessary for this PR. It would also complicate export logic / possibly other things, and would require removing the other `__overload__` mechanism first.